### DISCUSSION TITLE
Pocketbase record id

### DISCUSSION THREAD

**Discussion by lucius100** [pocketbase/pocketbase#3777] [Q&A] [answered]
"When working with my API automation app, I'm looking for a more user-friendly method to view records. Currently, the standard approach involves using a 15-character string in the (/api/collections/collectionIdOrName/records/recordId) to identify and view a specific record. However, I find it challenging to generate and manage these strings, especially when creating records and checking for duplicates.

Ideally, I'd like to use a 'code' instead of the lengthy string for viewing records. This would not only simplify the process but also streamline the task of checking for duplicate records before creation. Can you provide guidance on how to achieve this or if there's a more efficient method for viewing records based on a 'code'?"

**Comment by ganigeorgiev** [comment]
I don't understand what do you mean by _"especially when creating records and checking for duplicates"_.
You don't have to manually generate or check for uniquines of the record's id since it has a builtin unique constraint and generator. Or in other words - you can leave the `id` field empty and the value will be populated on the server-side on db insert.

At the moment it is not possible to change the length of the `id` field. Support for this is planned for the future as part of another broader task for a more "native" SQLite dev experience.

> Can you provide guidance on how to achieve this or if there's a more efficient method for viewing records based on a 'code'?"

You can register a custom route with [Go](https://pocketbase.io/docs/go-routing/#registering-new-routes) or the [JS pb_hooks](https://pocketbase.io/docs/js-routing/#registering-new-routes) and manually retrieve the record from there but I don't fully understand your use case and I don't feel confident recommending it.

If you want to retrieve a single record by a non-id value, you can use the existing [`List/Search` API](https://pocketbase.io/docs/api-records/#listsearch-records) with `?perPage=1`. If you are using the SDKs you can also make use of the `getFirstListItem()` helper method:

```js
await pb.collection('example').getFirstListItem('yourField = "something')
```

**Reply by lucius100** [reply]
Suppose I've already added a record with the following data:

```
{   "code": "news-001"
  }
```
  
  Now, if my application attempts to create another record with the same data:
  
  ```
{   "code": "news-001"
  }
```

It becomes a duplicate, even though the system generates a unique "id" for each record. The issue is that the data itself is duplicated. To address this, I want to implement a check to see if a record with the same data already exists before attempting to create a new one. If it does, the application can be programmed to skip the record creation process to avoid duplicates.

**Reply by ganigeorgiev** [reply, marked as answer]
You can add unique validation for any field (including combined fields or partial expression) by creating a unique index for them.

This can be done from the _Admin UI > Collections > Edit collection form > "Unique constraints and indexes"_ section:
![unique_screenshot](https://camo.githubusercontent.com/068c2f4d7fa52945b69fbbbcb887b521171f46a1fe387630519ee51fa80ae9c0/68747470733a2f2f692e696d6775722e636f6d2f6a617369434c4d2e706e67)

**Comment by lucius100** [comment]
> You can add unique validation for any field (including combined fields or partial expression) by creating a unique index for them.
> 
> This can be done from the _Admin UI > Collections > Edit collection form > "Unique constraints and indexes"_ section: ![unique_screenshot](https://camo.githubusercontent.com/068c2f4d7fa52945b69fbbbcb887b521171f46a1fe387630519ee51fa80ae9c0/68747470733a2f2f692e696d6775722e636f6d2f6a617369434c4d2e706e67)

Thanks, this solved my issue.
